From f6d5f18703a39be52ee04df5234f6ebd7ba1b9b7 Mon Sep 17 00:00:00 2001
From: Yves Orton <demerphq@gmail.com>
Date: Thu, 9 Feb 2023 17:02:35 +0100
Subject: [PATCH] lib/DBM/Deep.pm - apostrophe as package separator deprecated
 in 5.37.8

In 5.40.0 we will remove support for apostrophe as a package separator
entirely, and in 5.37.8 we marked it as deprecated so it will warn.

This patch removes its use from lib/DBM/Deep.pm and bumps the version
to 2.0017
---
 lib/DBM/Deep.pm  | 10 +++++-----
 lib/DBM/Deep.pod |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/lib/DBM/Deep.pm b/lib/DBM/Deep.pm
index 8a8ead7..61ff547 100644
--- a/lib/DBM/Deep.pm
+++ b/lib/DBM/Deep.pm
@@ -647,13 +647,13 @@ sub _warnif {
    # to avoid this hack. warnings.pm did not allow us to specify
    # exactly the call frame we want, so we have to look at the bitmask
    # ourselves.
-   if(  vec $bitmask, $warnings'Offsets{$_[0]}, 1,
-     || vec $bitmask, $warnings'Offsets{all}, 1,
+   if(  vec $bitmask, $warnings::Offsets{$_[0]}, 1,
+     || vec $bitmask, $warnings::Offsets{all}, 1,
      ) {
       my $msg = $_[1] =~ /\n\z/ ? $_[1] : "$_[1] at $file line $line.\n";
       die $msg
-       if  vec $bitmask, $warnings'Offsets{$_[0]}+1, 1,
-        || vec $bitmask, $warnings'Offsets{all}+1, 1;
+       if  vec $bitmask, $warnings::Offsets{$_[0]}+1, 1,
+        || vec $bitmask, $warnings::Offsets{all}+1, 1;
       warn $msg;
    }
   }
